cli/config/configfile/file_test.go

```diff
@@ -481,6 +481,57 @@ func TestLoadFromReaderWithUsernamePassword(t *testing.T) {
 	}
 }
 
+func TestGetAllCredentialsFromEnvironment(t *testing.T) {
+	t.Run("case=can fetch credentials from DOCKER_AUTH_CONFIG and underlying store", func(t *testing.T) {
+		configFile := New("filename")
+		exampleAuth := types.AuthConfig{
+			Username: "user",
+			Password: "pass",
+		}
+		configFile.AuthConfigs["example.com/foo"] = exampleAuth
+
+		envConfig := &ConfigFile{
+			AuthConfigs: map[string]types.AuthConfig{
+				"example.com/foo/env": {
```

Comment:
Wondering if we should use different examples here; I don't think we currently support multiple credentials per registry (hostname). From the test, it looks like that behavior is not consistent in all code paths, but perhaps using different hostnames instead of paths (as paths are stripped in various code paths) would make it more closely to how things are used (and not codify possibly missing "strip path" cases in tests) 🤔 


Could use (e.g.) `file.example.com` and `env.example.com`